docs/gds-supported-platform/tls-certificates.md

```diff
@@ -27,6 +32,76 @@ spec:
 
 > **Note:** cert-manager will need to be able to modify the DNS of the domains listed in the certificate in order to perform the DNS challenge. At the time of writing that only applies to the cluster domain.
 
+## Providing your own certificates
+
+If you want to provide your own certificate, you will need:
+
+ - a `SealedSecret` resource containing the certificate and key
+ - a `Gateway` resource defining the ingress domain
+
+We assume:
+
+ - you are in a namespace called `my-namespace`
+ - your certificate and key are in files called `my-cert.pem` and
+   `my-cert.key`
+ - you want to create a `SealedSecret` called `my-custom-cert` in a
+   file called `my-custom-cert.yaml`
+ - you are creating the certificate for the `sandbox` cluster
+ - the certificate is for `my-custom-domain.example.com`.
+
+To create a `SealedSecret` with your certificate, run:
+
+    kubectl create -n my-namespace secret generic my-custom-cert --dry-run --from-file=cert=my-cert.pem --from-file=key=my-cert.key --output yaml | gds sandbox seal --format yaml > my-custom-cert.yaml
+
+Here is an example of a `Gateway` that uses this certificate:
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: custom-certificate-gateway
+  namespace: my-namespace
+  annotations:
+    externaldns.k8s.io/namespace: my-namespace
+spec:
+  selector:
+    istio: my-namespace-ingressgateway
+  servers:
+  - port:
+      number: 443
+      name: https
+      protocol: HTTPS
+    tls:
+      mode: SIMPLE
+      credentialName: my-custom-cert
```

Comment:
```suggestion
      credentialName: my-custom-cert
      serverCertificate: sds
      privateKey: sds
```

Reply:
See https://istio.io/docs/reference/config/networking/v1alpha3/gateway/#Server-TLSOptions
`serverCertificate` and `privateKey` are required when TLS mode is `SIMPLE` or `MUTUAL`. Populating them with "sds" seems to be the convention when using SDS (as we are), but I guess they can be anything.

Reply:
Huh. Well despite what the docs say I tried this as-is (ie without those fields) and it worked fine, so maybe "REQUIRED" doesn't really mean "REQUIRED".

That said, I don't mind adding these fields to the docs.

Reply:
Interesting. I wonder if that got "fixed" between 1.1.4 and 1.1.12 then. I'm sure I tested that at some point and couldn't get it to work. :man_shrugging: I'm happy to go with either option in that case.